File: problems/3105-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.py
class Solution:
    def longestMonotonicSubarray(self, nums):
        n = len(nums)
        if n <= 1:
            return n

        max_len = 1
        for i in range(n):
            # Increasing
            curr_len = 1
            for j in range(i + 1, n):
                if nums[j] > nums[j - 1]:
                    curr_len += 1
                else:
                    break
            max_len = max(max_len, curr_len)

            # Decreasing
            curr_len = 1
            for j in range(i + 1, n):
                if nums[j] < nums[j - 1]:
                    curr_len += 1
                else:
                    break
            max_len = max(max_len, curr_len)
        return max_len

File: problems/3105-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int n = nums.size();
        if (n <= 1) {
            return n;
        }
        int maxLen = 1;
        for (int i = 0; i < n; ++i) {
            int increasingLen = 1;
            for (int j = i + 1; j < n; ++j) {
                if (nums[j] > nums[j - 1]) {
                    increasingLen++;
                } else {
                    break;
                }
            }
            maxLen = max(maxLen, increasingLen);

            int decreasingLen = 1;
            for (int j = i + 1; j < n; ++j) {
                if (nums[j] < nums[j - 1]) {
                    decreasingLen++;
                } else {
                    break;
                }
            }
            maxLen = max(maxLen, decreasingLen);
        }
        return maxLen;
    }
};